src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -380,29 +395,132 @@ internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<Functi
                 JToken durableTaskValue;
                 if (json.TryGetValue(DurableTask, StringComparison.OrdinalIgnoreCase, out durableTaskValue) && durableTaskValue != null)
                 {
-                    try
+                    var extensions = await _extensionsManager.GetExtensions();
+                    var durableTaskExtension = extensions.FirstOrDefault(ext => string.Equals(ext.Id, "Microsoft.Azure.WebJobs.Extensions.Durabletask", StringComparison.OrdinalIgnoreCase));
+                    if (durableTaskExtension == null)
                     {
-                        var kvp = (JObject)durableTaskValue;
-                        if (kvp.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+                        throw new InvalidOperationException(DurableNotInstalledErrorMsg);
+                    }
+
+                    var kvp = (JObject)durableTaskValue;
+                    if (durableTaskExtension.Version.StartsWith("1."))
+                    {
+                        if (this.HasDurableV2SpecificConfig(kvp))
                         {
-                            config.Add(HubName, nameValue.ToString());
+                            throw new InvalidOperationException(InvalidDurableV1ConfigErrorMsg);
                         }
-
-                        if (kvp.TryGetValue(DurableTaskStorageConnectionName, StringComparison.OrdinalIgnoreCase, out nameValue) && nameValue != null)
+                        return this.GetDurableV1Config(kvp);
+                    }
+                    else if (durableTaskExtension.Version.StartsWith("2."))
+                    {
+                        if (this.HasDurableV1SpecificConfig(kvp))
                         {
-                            config.Add(Connection, nameValue.ToString());
+                            throw new InvalidOperationException(InvalidDurableV2ConfigErrorMsg);
                         }
+                        return this.GetDurableV2Config(kvp);
+                    }
+                    else
+                    {
+                        throw new InvalidOperationException($"Cannot read host.json configuration for version {durableTaskExtension.Version} of durable task extension");
                     }
-                    catch (Exception)
+                }
+            }
+
+            return new Dictionary<string, string>();
+        }
+
+        private Dictionary<string, string> GetDurableV1Config(JObject durableHostConfig)
+        {
+            var config = new Dictionary<string, string>();
+            try
+            {
+                if (durableHostConfig.TryGetValue(HubName, StringComparison.OrdinalIgnoreCase, out JToken nameValue) && nameValue != null)
+                {
+                    config.Add(HubName, nameValue.ToString());
+                }
```

Comment:
For symmetry, I suggest you also apply the correct default in the v1 case as well.

Reply:
I'd rather let ScaleController do that defaulting for now, as hopefully we can remove this logic from the sync triggers side of things.